libparsec/crates/protocol/schema/anonymous_account_cmds/account_create_proceed.json5

```diff
@@ -59,6 +28,65 @@
         // virtually non-existent as long as the client generates a proper UUID.
         "status": "auth_method_id_already_exists"
       }
+    ],
+    "nested_types": [
+      {
+        "name": "AccountCreateStep",
+        "discriminant_field": "step",
+        "variants": [
+          {
+            "name": "Number0CheckCode",
+            "discriminant_value": "NUMBER_0_CHECK_CODE",
+            "fields": [
+              {
+                /// Code received by email following use of `account_create_send_validation_email`
+                "name": "code",
+                "type": "String"
+              }
+            ]
+          },
+          {
+            "name": "Number1Create",
+            "discriminant_value": "NUMBER_1_CREATE",
+            "fields": [
+              {
```

Comment:
validation_code is missing

Reply:
IMO, yes it's needed else we would need to have a statefull backend with cookies and stuff

Reply:
From what I understand we want to tweak the protocol to first check the code and then in another step proceed with the account creation to allow the front to ask separately for the code.

If we allow to do it all in one step, it kinda defeat that purpose ?

Reply:
> If we allow to do it all in one step, it kinda defeat that purpose ?

I mean the step is called `CHECK_CODE`, not `VALIDATE_CODE`.

by not providing the code in the second step, that require to ensure that `CHECK_CODE` was called before with a valid code.

For me, it seems simpler to just pass the code again

Reply:
> what happens if step0 is ok then step1 is called but with a different code ?

The API is stateless, so server would check the code and find that it's invalid.

> but what does step0 do if it does not validate / check that the provided code is the expected one?

It's expected that the check code step, check the code :shrug:  

> if we validate the code only in step1, there is no point in having step0 IMHO

It still as value as the client could send a request to validate that the user does enter the correct code before continuing further (i.e.: requesting the other elements).